src/Microsoft.ML.Core/Data/IDataView.cs

```diff
@@ -161,8 +159,25 @@ public interface IRow : ISchematized, ICounted
         /// <summary>
         /// Returns a value getter delegate to fetch the given column value from the row.
         /// This throws if the column is not active in this row, or if the type
-        /// <typeparamref name="TValue"/> differs from this row's schema's
-        /// <see cref="ISchema.GetColumnType(int)"/> on <paramref name="col"/>.
+        /// <typeparamref name="TValue"/> differs from this column's type.
+        /// </summary>
+        ValueGetter<TValue> GetGetter<TValue>(int col);
+    }
+
+    /// <summary>
+    /// A logical row. Every value of every column is retrievable, and immutable.
+    /// </summary>
+    public interface IStandaloneRow : ISchematized
+    {
+        /// <summary>
+        /// Puts a value of a column <paramref name="col"/> into <paramref name="value"/>.
+        /// This throws if the type <typeparamref name="TValue"/> differs from this column's type.
+        /// </summary>
+        void GetValue<TValue>(int col, ref TValue value);
```

Comment:
>GetValue<TValue>(int col, ref TValue value); [](start = 13, length = 44)

I suppose that it would be a bug for `GetGetter<>(col)(ref val` to result in a separate value from `GetValue<>(col ref val)`.

This *strongly suggests* that it would be better done via an extension method, rather than an implicit interface method. Otherwise we're just putting a bobby trap into our code.

Reply:
If separate implementations are somehow useful, might be useful to know why.

---
In reply to: [224531809](https://github.com/dotnet/machinelearning/pull/1167#discussion_r224531809) [](ancestors = 224531809)

Reply:
I think `GetValue` is the one I'd prefer to keep. `GetGetter` is useful for 'recombining' rows into other rows, but it doesn't add to the consumption layer.

---
In reply to: [224532997](https://github.com/dotnet/machinelearning/pull/1167#discussion_r224532997) [](ancestors = 224532997,224531809)

Reply:
Actually, I will for now remove the entire interface :) `MetadataRow` is the only implementation.

---
In reply to: [224543998](https://github.com/dotnet/machinelearning/pull/1167#discussion_r224543998) [](ancestors = 224543998,224532997,224531809)